nitransforms/surface.py

```diff
@@ -0,0 +1,124 @@
+# emacs: -*- mode: python-mode; py-indent-offset: 4; indent-tabs-mode: nil -*-
+# vi: set ft=python sts=4 ts=4 sw=4 et:
+### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+#
+#   See COPYING file distributed along with the NiBabel package for the
+#   copyright and license terms.
+#
+### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+"""Surface transforms."""
+
+import h5py
+import numpy as np
+import scipy.sparse as sparse
+
+from nitransforms.base import TransformBase
+
+
+class SurfaceTransform(TransformBase):
+    """Represents transforms between surface spaces."""
+
+    __slots__ = ("mat",)
+
+    def __init__(self, mat):
+        """Initialize the transform.
+
+        Parameters
+        ----------
+        mat : array-like, shape (nv1, nv2)
+            Sparse matrix representing the transform.
+        """
+        super().__init__()
+        if isinstance(mat, sparse.csr_array):
+            self.mat = mat
+        else:
+            self.mat = sparse.csr_array(mat)
+
+    def apply(self, x, inverse=False, normalize="element"):
+        """Apply the transform to surface data.
+
+        Parameters
+        ----------
+        x : array-like, shape (..., nv1)
```

Comment:
what is nv1? number of vertices?

Reply:
Yes, the number of vertices. I use `nv1` for the number of vertices of input, and `nv2` for the output, given they are likely to be different.